src/dialogs/voice-command-dialog/ha-voice-command-dialog.ts

```diff
@@ -211,18 +209,29 @@ export class HaVoiceCommandDialog extends LitElement {
   private _initRecognition() {
     this.recognition = new SpeechRecognition();
     this.recognition.interimResults = true;
-    this.recognition.lang = "en-US";
+    this.recognition.lang = this.hass.language;
 
-    this.recognition.onstart = () => {
+    this.recognition.addEventListener("start", () => {
       this.results = {
         final: false,
         transcript: "",
       };
-    };
-    this.recognition.onerror = (event) => {
+    });
+    this.recognition.addEventListener("nomatch", () => {
+      this._addMessage({
+        who: "user",
+        text: `<${this.hass.localize(
+          "ui.dialogs.voice_command.did_not_understand"
+        )}>`,
+        error: true,
+      });
+    });
+    this.recognition.addEventListener("error", (event) => {
+      // eslint-disable-next-line
+      console.error("Error recognizing text", event);
       this.recognition!.abort();
       // @ts-ignore
-      if (event.error !== "aborted") {
+      if (event.error !== "aborted" && event.error !== "no-speech") {
```

Comment:
not equal to `no-speech`? I would expect it should be `no-speech`?

Reply:
if the user clicked microphone but didn't speak, it feels weird to put up an error message? Prefer to just pretend nothing happened.

Reply:
Don't know, if your mic is not working or anything, it is weird that it doesn't do anything?

Reply:
If we want to do this, should we also remove it from `end` when the transcript is empty?

It seems that if there is sound but no speech is detected it will not error, but end without results, resulting in the `did_not_hear` error being displayed